### PR TITLE
Add HTTP resources

### LINKED ISSUE #11: Migrate servlets

### PR DESCRIPTION
**Pull request by babolivier**
This change introduces the user-facing HTTP resources that implement the API described here: https://github.com/matrix-org/matrix-content-scanner-python/blob/main/docs/api.md

Closes #11

### CODE REVIEW

File: matrix_content_scanner/servlets/thumbnail.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        self._scanner = content_scanner.scanner
+
+    async def on_GET(self, request: Request) -> Tuple[int, Union[bytes, JsonDict]]:
+        # mypy doesn't recognise request.postpath but it does exist and is documented.
+        media_path_bytes: bytes = b"/".join(request.postpath)  # type: ignore[attr-defined]
+        media_path = media_path_bytes.decode("ascii")
+        logutils.set_media_path_in_context(media_path)
+
+        # request.args stores all keys and values as bytes. However, we want them to be
+        # string going forward, so we convert them now.
```
Comment (squahtx) [inline matrix_content_scanner/servlets/thumbnail.py:42]: small typo:
```suggestion
        # request.args stores all keys and values as bytes. However, we want them to be
        # strings going forward, so we convert them now.
```

File: matrix_content_scanner/servlets/download.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def __init__(self, content_scanner: "MatrixContentScanner"):
+        super().__init__()
+        self._scanner = content_scanner.scanner
+        self._crypto_handler = content_scanner.crypto_handler
+
+    async def on_POST(self, request: Request) -> Tuple[int, Union[bytes, JsonDict]]:
+        media_path, metadata = get_media_metadata_from_request(
+            request, self._crypto_handler
+        )
```
Comment (squahtx) [inline matrix_content_scanner/servlets/download.py:59]: Can we move `logutils.set_media_path_in_context` out of `get_media_metadata_from_request`?
It's very unexpected for a function named `get_...` to have the side effect of configuring logging.

**Review by squahtx — CHANGES_REQUESTED** [review]
Largely looks good!

File: matrix_content_scanner/servlets/download.py (hunk 2 above)
Comment (babolivier) [reply matrix_content_scanner/servlets/download.py:59]: Sure. Though worth noting that `set_media_path_in_context` will be going away in https://github.com/matrix-org/matrix-content-scanner-python/issues/20 anyway.

**Review by squahtx — APPROVED** [review]
Thanks for making the changes. Looks good!

**Comment by squahtx** [comment]
Any idea why CI is unhappy?

**Comment by babolivier** [comment]
> Any idea why CI is unhappy?

Not yet but it looks unrelated to this PR as https://github.com/matrix-org/matrix-content-scanner-python/pull/31 is also unhappy for the same reasons. I'm having a look.

**Comment by babolivier** [comment]
The recent CI failure seems to be caused by the `setup-python` action (or whatever it uses to resolve Python versions) switching its definition of `3.x` from Python 3.10 to 3.11 overnight, and `python-olm` not having wheels for 3.11 on GitLab yet. I've temporarily set the CI to run 3.10 on this job in #31 to fix it, and have pinged people internally.

**Pull request merged**